src/main/java/org/opentripplanner/graph_builder/module/GtfsFeedId.java

```diff
@@ -99,12 +116,23 @@ protected String cleanId(String id) {
          * @return A GtfsFeedId
          */
         public GtfsFeedId build() {
+            boolean autoGenerated = false;
             id = cleanId(id);
             if (id == null || id.trim().length() == 0) {
                 id = String.valueOf(FEED_ID_COUNTER);
+                autoGenerated = true;
+            }
+            if (seenIds.contains(id)) {
+                throw new RuntimeException(
+                    String.format(
+                        "A duplicate Feed ID of `%s` was encountered when loading GTFS data!",
+                        id
+                    )
+                );
             }
+            seenIds.add(id);
```

Comment:
Since we're touching it, I think this block could use a comment or two about auto-generating the feed id. Also, minor syntax thing, if you wanted to, you could just check the return value of `seenIds.add(id)` and throw the exception above if `false` if you wanted to remove a line of code (I'd recommend adding a comment though).